src/rules/no-unused-modules.js

```diff
@@ -174,17 +294,36 @@ const isNodeModule = (path) => (/\/(node_modules)\//).test(path);
 const resolveFiles = (src, ignoreExports, context) => {
   const extensions = Array.from(getFileExtensions(context.settings));
 
-  const srcFileList = listFilesToProcess(src, extensions);
+  const srcFileList = listFilesToProcess(src, extensions, context);
 
   // prepare list of ignored files
-  const ignoredFilesList = listFilesToProcess(ignoreExports, extensions);
+  const ignoredFilesList = listFilesToProcess(
+    ignoreExports,
+    extensions,
+    context,
+  );
+
+  // The modern api will return a list of file paths, rather than an object
```

Comment:
The shape of the legacy `listFilesToProcess` was an array of objects consisting of `filename` and `ignored`.  
![image](https://github.com/user-attachments/assets/f9234851-7499-4821-844a-331890e834af)

Whereas the new api just returns a list of filepaths
![image](https://github.com/user-attachments/assets/7c2ba890-b67f-42a3-85be-38209da1bb1d)